compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -246,7 +238,7 @@ impl<'tcx> ConstToPat<'tcx> {
                         FloatPattern,
                     );
                 }
-                PatKind::Constant { value: cv }
+                return Err(FallbackToConstRef);
```

Comment:
This doesn't change anything, as we have been using PartialEq for floats already.